app/controllers/user_sessions_controller.rb

```diff
@@ -6,75 +6,132 @@ def new
   end
 
   def create
-    params[:user_session][:username] = params[:openid] if params[:openid] # second runthrough must preserve username
-    username = params[:user_session][:username] if params[:user_session]
-    @user = User.find_by(username: username)
+    auth = request.env['omniauth.auth']
```

Comment:
So here, are you sure you want the `request.env[]` vs. a Ruby env variable like `ENV['OMNIAUTH_AUTH']`? Just checking.

Reply:
``request.env['omniauth.auth']`` is the hash returned by the provider by the callback url whereas ``ENV['OMNIAUTH_AUTH']`` has '``OMNIAUTH_AUTH' ``which stores the value of ``app_id`` of the app being created by you. So, I am sure the other way around will not work. Also, I checked this doesn't work.